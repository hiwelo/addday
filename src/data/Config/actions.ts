import { ActionsUnion, ConfigState } from '../types';

/**
 * Resets the current user for this application
 */
function setSentryEnrollment(
  updatedEnrollmentStatus: ConfigState['sentryEnrollment'],
): ActionsUnion {
  return {
    type: 'SENTRY_ENROLLMENT',
    payload: updatedEnrollmentStatus,
  };
}

export { setSentryEnrollment };
