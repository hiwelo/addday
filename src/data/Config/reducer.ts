import { ConfigState, ConfigAction } from './types';

const INITIAL_STATE: ConfigState = {
  sentryEnrollment: false,
};

function configReducer(
  state = INITIAL_STATE,
  action: ConfigAction,
): ConfigState {
  const { type, value } = action;

  switch (type) {
    case 'SENTRY_ENROLLMENT':
      return {
        ...state,
        sentryEnrollment: value.sentryEnrollment,
      };

    default:
      return state;
  }
}

export default configReducer;
