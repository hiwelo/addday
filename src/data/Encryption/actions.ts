import { ActionsUnion } from '../types';

/**
 * Fetches the encryption key
 */
function getEncryptionKey(): ActionsUnion {
  return {
    type: 'GET_ENCRYPTION_KEY',
  };
}

/**
 * Saves a new encryption key
 */
function setEncryptionKey(key: string): ActionsUnion {
  return {
    type: 'SET_ENCRYPTION_KEY',
    payload: key,
  };
}

export { getEncryptionKey, setEncryptionKey };
