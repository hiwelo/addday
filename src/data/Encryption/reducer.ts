import { ActionsUnion, EncryptionState } from '../types';

function encryptionReducer(
  state: EncryptionState = { key: null },
  { type }: ActionsUnion,
): EncryptionState {
  switch (type) {
    case 'GET_ENCRYPTION_KEY':
      return state;

    case 'SET_ENCRYPTION_KEY':
      return {
        ...state,
        key: 'test',
      };

    default:
      return state;
  }
}

export default encryptionReducer;
