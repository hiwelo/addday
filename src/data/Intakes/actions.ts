import { Intake } from '../../models/Intake';
import { ActionsUnion } from '../types';

export function clearIntakes(): ActionsUnion {
  return {
    type: `CLEAR_INTAKES_ACTION`,
  };
}

export function deleteIntake(intakeId: Intake['id']): ActionsUnion {
  return {
    type: `DELETE_INTAKE_ACTION`,
    payload: intakeId,
  };
}

export function setIntake(intake: Intake): ActionsUnion {
  return {
    type: `SET_INTAKE_ACTION`,
    payload: intake,
  };
}
