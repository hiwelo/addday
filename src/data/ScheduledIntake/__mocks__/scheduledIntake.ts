import moment from 'moment';

import { WeeklySchedule } from '../../../models/WeeklySchedule';

export const mockedMoment = moment.utc('01/01/2020 18:00', 'dd/mm/yyyy HH:mm');

export const mockedWeeklySchedule: WeeklySchedule = {
  monday: true,
  tuesday: false,
  wednesday: false,
  thursday: true,
  friday: true,
  saturday: false,
  sunday: false,
};

export const mockedScheduledIntake = {
  days: mockedWeeklySchedule,
  id: 'testScheduledIntake',
  moment: mockedMoment,
  medications: {},
};
