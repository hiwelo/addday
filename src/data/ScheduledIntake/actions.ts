import {
  ScheduledIntake,
  ScheduledMedication,
} from '../../models/ScheduledIntake';
import { ActionsUnion } from '../types';

export function clearScheduledIntakes(): ActionsUnion {
  return {
    type: `CLEAR_SCHEDULED_INTAKES_ACTION`,
  };
}

export function clearScheduledMedicationIntakes(
  scheduledIntakeId: ScheduledIntake['id'],
): ActionsUnion {
  return {
    type: `CLEAR_SCHEDULED_MEDICATION_INTAKES_ACTION`,
    payload: scheduledIntakeId,
  };
}

export function deleteScheduledIntake(
  scheduledIntakeId: ScheduledIntake['id'],
): ActionsUnion {
  return {
    type: `DELETE_SCHEDULED_INTAKE_ACTION`,
    payload: scheduledIntakeId,
  };
}

export function deleteScheduledMedicationIntake(
  scheduledIntakeId: ScheduledIntake['id'],
  scheduledMedicationId: ScheduledMedication['medicationId'],
): ActionsUnion {
  return {
    type: `DELETE_SCHEDULED_MEDICATION_INTAKE_ACTION`,
    payload: {
      scheduledIntakeId,
      scheduledMedicationId,
    },
  };
}

export function setScheduledIntake(
  scheduledIntake: ScheduledIntake,
): ActionsUnion {
  return {
    type: `SET_SCHEDULED_INTAKE_ACTION`,
    payload: scheduledIntake,
  };
}

export function setScheduledMedicationIntake(
  scheduledIntakeId: ScheduledIntake['id'],
  scheduledMedicationIntake: ScheduledMedication,
): ActionsUnion {
  return {
    type: `SET_SCHEDULED_MEDICATION_INTAKE_ACTION`,
    payload: {
      scheduledIntakeId,
      scheduledMedicationIntake,
    },
  };
}
