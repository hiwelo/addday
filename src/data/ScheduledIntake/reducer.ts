import { ActionsUnion, ScheduledIntakeState } from '../types';

const INITIAL_STATE: ScheduledIntakeState = {};

function scheduledIntakeReducer(
  state: ScheduledIntakeState = INITIAL_STATE,
  action: ActionsUnion,
): ScheduledIntakeState {
  switch (action.type) {
    case `CLEAR_SCHEDULED_INTAKES_ACTION`:
    case `RESET_APP`:
      return INITIAL_STATE;

    case `CLEAR_SCHEDULED_MEDICATION_INTAKES_ACTION`: {
      const scheduledIntake = state[action.payload];

      if (!scheduledIntake) {
        throw new Error('The requested scheduled intake was not found.');
      }

      return {
        ...state,
        [scheduledIntake.id]: {
          ...scheduledIntake,
          medications: {},
        },
      };
    }

    case `DELETE_SCHEDULED_INTAKE_ACTION`: {
      delete state[action.payload];

      return state;
    }

    case `DELETE_SCHEDULED_MEDICATION_INTAKE_ACTION`: {
      const scheduledIntake = state[action.payload.scheduledIntakeId];

      if (!scheduledIntake) {
        throw new Error('The requested scheduled intake was not found.');
      }

      delete scheduledIntake.medications[action.payload.scheduledMedicationId];

      return {
        ...state,
        [scheduledIntake.id]: scheduledIntake,
      };
    }

    case `SET_SCHEDULED_INTAKE_ACTION`:
      return {
        ...state,
        [action.payload.id]: action.payload,
      };

    case `SET_SCHEDULED_MEDICATION_INTAKE_ACTION`: {
      const scheduledIntake = state[action.payload.scheduledIntakeId];

      if (!scheduledIntake) {
        throw new Error('The requested scheduled intake was not found.');
      }

      return {
        ...state,
        [scheduledIntake.id]: {
          ...scheduledIntake,
          medications: {
            ...scheduledIntake.medications,
            [action.payload.scheduledMedicationIntake.medicationId]:
              action.payload.scheduledMedicationIntake,
          },
        },
      };
    }

    default:
      return state;
  }
}

export default scheduledIntakeReducer;
