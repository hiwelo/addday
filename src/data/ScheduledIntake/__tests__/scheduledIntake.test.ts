import { mockedScheduledIntake } from '../__mocks__/scheduledIntake';
import * as actions from '../actions';
import reducer from '../reducer';

describe('Scheduled Intakes', () => {
  it('allows you to add a new scheduled intake', () => {
    const action = actions.setScheduledIntake(mockedScheduledIntake);
    const state = reducer(undefined, action);

    expect(action).toMatchSnapshot();
    expect(state).toMatchSnapshot();
  });

  it('allows you to update an existing scheduled intake', () => {
    const action = actions.setScheduledIntake(mockedScheduledIntake);
    const state = reducer(undefined, action);
    const updateAction = actions.setScheduledIntake({
      ...mockedScheduledIntake,
      medications: {
        test: {
          medicationId: 'test',
          units: 2,
        },
      },
    });
    const updatedState = reducer(state, updateAction);

    expect(updateAction).toMatchSnapshot();
    expect(updatedState).toMatchSnapshot();
  });

  it('allows you to delete an existing scheduled intake', () => {
    const action = actions.setScheduledIntake(mockedScheduledIntake);
    const state = reducer(undefined, action);
    const updateAction = actions.deleteScheduledIntake(
      mockedScheduledIntake.id,
    );
    const updatedState = reducer(state, updateAction);

    expect(updateAction).toMatchSnapshot();
    expect(updatedState).toMatchSnapshot();
  });

  it('allows you to clear all existing scheduled intake', () => {
    const action1 = actions.setScheduledIntake(mockedScheduledIntake);
    const action2 = actions.setScheduledIntake({
      ...mockedScheduledIntake,
      id: 'test2',
    });
    const state = reducer(undefined, action1);
    const updatedState = reducer(state, action2);
    const clearAction = actions.clearScheduledIntakes();
    const clearedState = reducer(updatedState, clearAction);

    expect(clearAction).toMatchSnapshot();
    expect(clearedState).toMatchSnapshot();
  });
});
