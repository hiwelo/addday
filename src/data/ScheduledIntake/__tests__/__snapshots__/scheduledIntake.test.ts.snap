// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scheduled Intakes allows you to add a new scheduled intake 1`] = `
Object {
  "payload": Object {
    "days": Object {
      "friday": true,
      "monday": true,
      "saturday": false,
      "sunday": false,
      "thursday": true,
      "tuesday": false,
      "wednesday": false,
    },
    "id": "testScheduledIntake",
    "medications": Object {},
    "moment": "2020-01-01T18:00:00.000Z",
  },
  "type": "SET_SCHEDULED_INTAKE_ACTION",
}
`;

exports[`Scheduled Intakes allows you to add a new scheduled intake 2`] = `
Object {
  "testScheduledIntake": Object {
    "days": Object {
      "friday": true,
      "monday": true,
      "saturday": false,
      "sunday": false,
      "thursday": true,
      "tuesday": false,
      "wednesday": false,
    },
    "id": "testScheduledIntake",
    "medications": Object {},
    "moment": "2020-01-01T18:00:00.000Z",
  },
}
`;

exports[`Scheduled Intakes allows you to clear all existing scheduled intake 1`] = `
Object {
  "type": "CLEAR_SCHEDULED_INTAKES_ACTION",
}
`;

exports[`Scheduled Intakes allows you to clear all existing scheduled intake 2`] = `Object {}`;

exports[`Scheduled Intakes allows you to delete an existing scheduled intake 1`] = `
Object {
  "payload": "testScheduledIntake",
  "type": "DELETE_SCHEDULED_INTAKE_ACTION",
}
`;

exports[`Scheduled Intakes allows you to delete an existing scheduled intake 2`] = `Object {}`;

exports[`Scheduled Intakes allows you to update an existing scheduled intake 1`] = `
Object {
  "payload": Object {
    "days": Object {
      "friday": true,
      "monday": true,
      "saturday": false,
      "sunday": false,
      "thursday": true,
      "tuesday": false,
      "wednesday": false,
    },
    "id": "testScheduledIntake",
    "medications": Object {
      "test": Object {
        "medicationId": "test",
        "units": 2,
      },
    },
    "moment": "2020-01-01T18:00:00.000Z",
  },
  "type": "SET_SCHEDULED_INTAKE_ACTION",
}
`;

exports[`Scheduled Intakes allows you to update an existing scheduled intake 2`] = `
Object {
  "testScheduledIntake": Object {
    "days": Object {
      "friday": true,
      "monday": true,
      "saturday": false,
      "sunday": false,
      "thursday": true,
      "tuesday": false,
      "wednesday": false,
    },
    "id": "testScheduledIntake",
    "medications": Object {
      "test": Object {
        "medicationId": "test",
        "units": 2,
      },
    },
    "moment": "2020-01-01T18:00:00.000Z",
  },
}
`;
